eq-author/src/components/CommentsPanel/index.js

```diff
@@ -145,57 +277,105 @@ const CommentsPanel = ({
     return <Error>Oops! Something went wrong</Error>;
   }
 
-  const commentsArray = get(data, "page.comments", []);
+  const displayComments = commentsArray.map((item, index) => {
+    const editCommentId = `name-${index}`;
+
+    return (
+      <CommentSection key={item.id}>
+        <CommentHeaderContainer>
+          <AvatarWrapper>
+            <AvatarOuter>
+              <AvatarInner>{getInitials(item.user.name)}</AvatarInner>
+            </AvatarOuter>
+          </AvatarWrapper>
+          <NameWrapper>
+            <FlexLabel>{item.user.displayName}</FlexLabel>
+            <DateField>{moment(item.createdTime).calendar()}</DateField>
+          </NameWrapper>
+          <EditButtonIcon
+            disabled={item.user.id !== myId}
+            onClick={() => handleEdit(index)}
+            data-test={`btn-edit-comment-${index}`}
+          />
+          <DeleteComment
+            disabled={item.user.id !== myId}
+            onClick={() => handleDelete(item)}
+            data-test={`btn-delete-comment-${index}`}
+          />
+        </CommentHeaderContainer>
+        <StyledTextArea
+          id={item.id}
+          disabled={activeComment !== index}
+          defaultValue={item.commentText}
+          name={editCommentId}
+          type="text"
+          rows={2}
+          onChange={onChangeEditState}
+          data-test={`edit-comment-txtArea-${index}`}
+        >
+          {}
```

Comment:
Is this needed?

Reply:
used in test for comment update
eq-author/src/components/CommentsPanel/index.test.js